chia/full_node/full_node.py

```diff
@@ -557,10 +557,17 @@ def _close(self):
         self._shut_down = True
         if self._init_weight_proof is not None:
             self._init_weight_proof.cancel()
-        if self.blockchain is not None:
+        # blockchain is created in _start and in certain cases it may not exist here during _close
+        try:
             self.blockchain.shut_down()
-        if self.mempool_manager is not None:
+        except AttributeError:
```

Comment:
I would have expected something like this:

```
if self.blockchain:
    self.blockchain.shut_down()
```

instead. This would swallow and suppress `AttributeError`s being throw out of the `shut_down()` functions.

Reply:
Changed to hasattr